src/main/java/com/jcabi/ssh/SSH.java

```diff
@@ -215,7 +237,15 @@ protected Session session() throws IOException {
                 CharEncoding.UTF_8
             );
             jsch.setHostKeyRepository(new EasyRepo());
-            jsch.addIdentity(file.getAbsolutePath());
+            if (this.passphrase == null) {
```

Comment:
@longtimeago Could you be more specific please? A puzzle? I don't really know what you want me to change.